libraries/standard/http/utest/http_utest.c

```diff
@@ -638,3 +767,220 @@ void test_Http_AddRangeHeader_With_Max_INT32_Range_Values( void )
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, retCode );
+    tearDown();
+    testResponse.bufferLen = strlen( pTestResponse );
+    retCode = HTTPClient_ReadHeader( &testResponse,
+                                     ( const uint8_t * ) "Header",
+                                     strlen( "Header" ),
+                                     &pValueLoc,
+                                     NULL );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, retCode );
+}
+
+/**
+ * @brief Test when requested header is not present in response.
+ */
+void test_Http_ReadHeader_Header_Not_In_Response( void )
+{
+    /* Add expectations for http_parser dependencies. */
+    http_parser_init_ExpectAnyArgs();
+    http_parser_settings_init_ExpectAnyArgs();
+
+    /* Configure the http_parser_execute mock. */
+    invokeHeaderCompleteCallback = false;
+    parserErrNo = HPE_OK;
+    http_parser_execute_ExpectAnyArgsAndReturn( strlen( pTestResponse ) );
+
+    /* Call the function under test. */
+    testResponse.bufferLen = strlen( pTestResponse );
+    retCode = HTTPClient_ReadHeader( &testResponse,
+                                     ( const uint8_t * ) HEADER_NOT_IN_BUFFER,
+                                     strlen( HEADER_NOT_IN_BUFFER ),
+                                     &pValueLoc,
+                                     &valueLen );
+    TEST_ASSERT_EQUAL( HTTP_HEADER_NOT_FOUND, retCode );
+}
+
+/**
+ * @brief Test with invalid HTTP responses.
+ */
+void test_Http_ReadHeader_With_Invalid_Responses()
+{
+    /* Case 1: Test when invalid response only contains the header field for the requested header. */
+    const char * pResponseWithoutValue = "HTTP/1.1 200 OK\r\n"
+                                         "test-header0: test-value0\r\n"
+                                         "test-header1:";
+
+    /* Add expectations for http_parser init dependencies. */
+    http_parser_init_ExpectAnyArgs();
+    http_parser_settings_init_ExpectAnyArgs();
+
+    /* Configure the http_parser_execute mock. */
+    pExpectedBuffer = pResponseWithoutValue;
+    expectedBufferSize = strlen( pResponseWithoutValue );
+    invokeHeaderFieldCallback = true;
+    pFieldLocToReturn = &pTestResponse[ headerFieldInRespLoc ];
+    fieldLenToReturn = headerFieldInRespLen;
+    http_parser_execute_ExpectAnyArgsAndReturn( strlen( pResponseWithoutValue ) );
+
+    /* Call the function under test. */
+    testResponse.pBuffer = ( uint8_t * ) &pResponseWithoutValue[ 0 ];
+    testResponse.bufferLen = strlen( pResponseWithoutValue );
+    retCode = HTTPClient_ReadHeader( &testResponse,
+                                     ( const uint8_t * ) HEADER_IN_BUFFER,
+                                     strlen( HEADER_IN_BUFFER ),
+                                     &pValueLoc,
+                                     &valueLen );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_RESPONSE, retCode );
+
+    /* Test when the invalid response does not contain the requested header.
```

Comment:
This should probably go into a separate test case function.
test_Http_ReadHeader_Only_Header_Field_Found()
test_Http_ReadHeader_No_Headers_Complete_Ending()

Reply:
Good suggestion.

Reply:
Updated